core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -120,38 +119,32 @@ private ReadWriteLock getPageLock(PageId pageId) {
   }
 
   @Override
-  public void put(PageId pageId, byte[] page) throws IOException {
+  public boolean put(PageId pageId, byte[] page) throws IOException {
     PageId victim = null;
+    long victimPageSize = 0;
+    boolean enoughSpace;
 
     ReadWriteLock pageLock = getPageLock(pageId);
     try (LockResource r = new LockResource(pageLock.writeLock())) {
-      boolean alreadyCached;
-      boolean needEvict = false;
       try (LockResource r2 = new LockResource(mMetaLock.writeLock())) {
-        alreadyCached = mMetaStore.hasPage(pageId);
-        if (!alreadyCached) {
-          needEvict = mPageStore.size() + 1 > mCacheSize;
-          if (needEvict) {
-            victim = mEvictor.evict();
-          } else {
-            mMetaStore.addPage(pageId);
-          }
+        if (mMetaStore.hasPage(pageId)) {
```

Comment:
I changed the behavior here a bit: if page to insert exists, it returns false now